knime_extension/src/nodes/conversion.py

```diff
@@ -718,19 +765,21 @@ def execute(self, exec_context: knext.ExecutionContext, input_table):
             geolocator.geocode,
             min_delay_seconds=self.geocoding_service_settings.min_delay_seconds,
         )
-        df["latitude"] = df[self.address_col].apply(lambda x: geocode(x).latitude)
-        df["longitude"] = df[self.address_col].apply(lambda x: geocode(x).longitude)
-        gdf = gp.GeoDataFrame(
-            df, geometry=gp.points_from_xy(df.longitude, df.latitude), crs="EPSG:4326"
-        )
 
-        gdf.drop(columns=["latitude", "longitude"], inplace=True)
-        # result_col_name = knut.get_unique_column_name(
-        #     self.name, input_table.schema
-        # )
-        # gdf.rename(columns={"geometry": result_col_name}, inplace=True)
+        tmp_col = knut.get_unique_column_name("__location__", input_table.schema)
```

Comment:
I introduced a temporary column to call the expensive geocode() method only once instead of twice per row. If you have another idea how to get rid of the temp column all together e.g. by getting the data and converting it into a Point within one method this would be even better :-)

Reply:
We removed that tmp_col after all the work was done. It looks fine to me. we need that tmp_col otherwise we will call the geocode() several times